app/src/main/java/cm/aptoide/pt/util/ReferrerUtils.java

```diff
@@ -139,10 +139,15 @@ public static void extractReferrer(SearchAdResult searchAdResult, final int retr
               if (broadcastReferrer) {
                 broadcastReferrer(packageName, referrer, context);
               } else {
-                StoredMinimalAdAccessor storedMinimalAdAccessor = AccessorFactory.getAccessorFor(
-                    ((AptoideApplication) context.getApplicationContext()
-                        .getApplicationContext()).getDatabase(), StoredMinimalAd.class);
-                storedMinimalAdAccessor.insert(adMapper.map(searchAdResult, referrer));
+                RoomStoredMinimalAdPersistence roomStoredMinimalAdPersistence =
+                    new RoomStoredMinimalAdPersistence(
+                        ((AptoideApplication) context.getApplicationContext()).getAptoideDatabase()
+                            .storeMinimalAdDAO());
+
+                Completable.fromAction(() -> roomStoredMinimalAdPersistence.insert(
+                    adMapper.map(searchAdResult, referrer)))
+                    .subscribeOn(Schedulers.io())
+                    .subscribe();
```

Comment:
why use a subscription here?

Reply:
For two reasons:
- to avoid creating a new thread (i think this would be uglier than doing it the rx way).
- Because i had a log here to test the insertion and then forgot to change this.